[FEATURE]: Handle creating a service file for croncatd

SeedyROM opened the pull request:
**Create feature that was mentioned in #32.**
- [x] Change verbiage for `daemon` to `setup-service`, to make it slightly more descriptive of what's going on?
- [x] Use the command `croncatd --chain-id <chain-id> setup-service` to create system services in a relative folder called `system-services`.
    - For instance `croncatd --chain-id local setup-service` creates a file at path `./system-services/croncatd-local.service`.
- [x] Fix strange use of croncatd chain_id being optional but also not optional?
- [x] Refactor system into it's own sub module folder.
- [x] Define the `ServiceDaemon` struct to help build system service files.
- [x] Link the newly created service into the system via a sudo command.

## Resources:
- Calling sudo and capturing stdin: https://users.rust-lang.org/t/how-to-execute-a-root-command-on-linux/50066/8

SeedyROM committed:
Handle creating a service file for croncatd

- Use the command `croncatd --chain-id <chain-id> daemonize`
- Fix strange use of croncatd chain_id being optional but also not
  optional?
- Refactor system into it's own sub module folder.
- Define the DaemonService struct to help build system service files.

SeedyROM committed:
:wrench: Fix warnings for unused imports

SeedyROM committed:
:wrench: Fixed typos and extra whitespace, named format params for service file creation instead of indexes

SeedyROM committed:
:hammer: Remove excess whitespace from croncatd main fn doc

SeedyROM commented:
Idea of note (handling only systemd?): https://github.com/CronCats/croncat-rs/issues/32#issuecomment-1253461479

SeedyROM committed:
:hammer: Fix missing = in banner.txt

SeedyROM commented:
ProTip: Don't use github mobile to try and link PRs with issues, :'( 

Sorry for the spam!

SeedyROM committed:
:hammer: Fix structopt doc for daemonize

mikedotexe commented:
Thanks @SeedyROM I am all in favor of your suggestion of `create-service`
Feel free to hit the 👍🏼 or 👎🏼 emoji on it here

SeedyROM committed:
:tada: Add missing docs for top-level modules, clean up opts.rs for the subcommand struct

SeedyROM committed:
:hammer: Rename Daemonize to SetupService, fix doc string for option to include that it's a systemd service

SeedyROM committed:
:hammer: Properly link the service to systemd.

- Rename ServiceDaemon, DaemonService
- Create a service file locally and link it to the systemd services.
- Print a nice helper message if frills are enabled.

SeedyROM commented:
@mikedotexe @TrevorJTClarke I went with `setup-service`, but this is not a huge change in the future. Any thoughts?

SeedyROM committed:
:hammer: Update README.md help text

SeedyROM committed:
:rotating_light: Remove struct docs for Opts and Command, it broke the help output about

SeedyROM committed:
:hammer: Update broken README.md help output

TrevorJTClarke commented on croncatd/banner.txt:
```diff
@@ -3,4 +3,4 @@
  ( o.o ) | (__| | | (_) | | | | (_| (_| | || (_| |
   > ^ <   \___|_|  \___/|_| |_|\___\__,_|\__\__,_|
-==================================================
+===================================================
```
This.
Normalize comment incrementing

mikedotexe replied:
Normalize adding some equality to this world.

Buckram123 commented on README.md:
```diff
@@ -30,13 +30,33 @@ croncatd 0.1.0
 
 USAGE:
-    croncatd [FLAGS]
+    croncatd [FLAGS] --chain-id <chain-id> <SUBCOMMAND>
```
Thanks for updating the readme <3

Buckram123 commented on croncat/src/system/service.rs:
```diff
@@ -0,0 +1,125 @@
+                    [Install]
+                    WantedBy=multi-user.target
+                "},
+                chain_id = chain_id,
+                user = user,
+                service_dir = full_service_dir_path.to_str().ok_or_else(|| eyre!(
+                    "Could not convert daemon service directory path to string",
+                ))?,
+                exe_path = std::env::current_exe()?.to_str().ok_or_else(|| eyre!(
+                    "Could not convert daemon service executable path to string",
```
that looks so great

Buckram123 commented on croncat/src/store/mod.rs:
```diff
@@ -1,3 +1,5 @@
+//! Helpers to handle storing and retrieving data from the local filesystem.
+
```
🔥

Buckram123 commented on croncat/src/config/mod.rs:
```diff
@@ -23,13 +24,9 @@ impl ChainConfig {
-        let config_file = &format!("config.{}.yaml", ch_id);
-        let config_override_file = &format!("config.{}.override.yaml", ch_id);
+    pub async fn new(chain_id: &String) -> Result<Self, Report> {
+        let config_file = &format!("config.{}.yaml", chain_id);
+        let config_override_file = &format!("config.{}.override.yaml", chain_id);
```
thanks for fixing it

Buckram123 reviewed (approved):
Great job! 🧨🧨🧨
And thanks for updating the `README`!

Buckram123 commented:
Funny it doesn't count my approve, cause I dont have an right access to this repo

SeedyROM commented:
> Funny it doesn't count my approve, cause I dont have an right access to this repo

It counts for me hahaha.
